src/rust/src/lib.rs

```diff
@@ -2,7 +2,86 @@
 // 2.0, and the BSD License. See the LICENSE file in the root of this repository
 // for complete details.
 
+/// Returns the value of the input with the most-significant-bit copied to all
+/// of the bits.
+fn duplicate_msb_to_all(a: u8) -> u8 {
+    0u8.wrapping_sub(a >> 7)
+}
+
+/// This returns 0xFF if a < b else 0x00, but does so in a constant time
+/// fashion.
+fn constant_time_lt(a: u8, b: u8) -> u8 {
+    // Derived from:
+    // https://github.com/openssl/openssl/blob/OpenSSL_1_1_1i/include/internal/constant_time.h#L120
+    duplicate_msb_to_all(a ^ ((a ^ b) | (a.wrapping_sub(b) ^ b)))
+}
+
+#[pyo3::prelude::pyfunction]
+fn check_pkcs7_padding(data: &[u8], block_len: u8) -> bool {
+    let mut mismatch = 0;
+    let pad_size = data[(block_len - 1) as usize];
+    for i in 0..block_len {
+        let mask = constant_time_lt(i, pad_size);
+        let b = data[(block_len - 1 - i) as usize] as u8;
```

Comment:
It seems like `as u8` is unnecessary here as (as per `cargo check` with the `trivial_numeric_casts` check enabled)